Anshul/cache features (#1)
AnshulBhl/azure-powershell: src/RedisCache/RedisCache/help/New-AzRedisCache.md

          Location           : Central US
          Name               : mycache
          Type               : Microsoft.Cache/Redis
          HostName           : mycache.redis.cache.windows.net
          Port               : 6379
          ProvisioningState  : creating
          SslPort            : 6380
          RedisConfiguration : {[maxmemory-policy, allkeys-random], [maxclients, 7500], [maxmemory-reserved, 200],
                                [maxfragmentationmemory-reserved, 300]...} 
          EnableNonSslPort   : False
          RedisVersion       : 4.0.14
          Size               : 6GB
          Sku                : Premium
          Tag                : {}
          Zone               : {1, 2}
```
This command creates Redis cache instance in mutliple zones.

### Example 4: Create a Virtual Network enable Cache
   
Requirements for creating Virtual Network enable cache:-
    1. Create the virtual network in same resource group in which you want to create your redis cache.
    2. You will need SubnetID for VNET enable cache. Syntax of SubnetID is given below.
 

> You can create virtual network from powershell command [New-AzVirtualNetwork](https://docs.microsoft.com/en-us/powershell/module/az.network/new-azvirtualnetwork?view=azps-6.4.0). 

Reviewer: Shall we use relative path for hyperlink?

          EnableNonSslPort   : False
          RedisVersion       : 4.0.14
          Size               : 6GB
          Sku                : Premium
          SubnetId           : /subscriptions/a559b6fd-3a84-40bb-a450-b0db5ed37dfe/resourceGroups/mygroup/providers/Microsoft.Network/virtualNetworks/MyNet/subnets/MySubnet
          StaticIP           : 10.0.0.4  
          Tag                : {}
          Zone               : []
```

### Example 5: Configure data persistence for a Premium Azure Cache for Redis

Persistence writes Redis data into an Azure Storage account that you own and manage. So before configuring data persistence you need to have [storage account] (https://docs.microsoft.com/en-us/azure/storage/common/storage-account-create?tabs=azure-powershell) in same resource group. Choose a storage account in the same region and subscription as the cache, and a Premium Storage account is recommended because premium storage has higher throughput.

For setting up data persistence, there is need to specifies Redis Configuration settings.   
For RDB back up enabled :-
    - rdb-backup-enabled (Set true or false)

>     - rdb-storage-connection-string (Go to your **Storage account** on portal. In left coloumn select **Access keys**. Then click on **Show keys** and copy any **connection string** from there.)

Reviewer: It is better to redirect to some storage account documentation for steps to get the connection string.
We can also add a command example to get the connection string.

    - rdb-backup-frequency (Set a backup interval in minutes. You can only choose from - 15, 30, 60, 360, 720 and 1440 minutes.)  
    
```

> PS C:\>New-AzRedisCache -ResourceGroupName "MyGroup" -Name "MyCache" -Location "Central US" -Size P1 -Sku "Premium" -RedisConfiguration @{"rdb-backup-enabled" = "true"; "rdb-storage-connection-string" = "DefaultEndpointsProtocol=https;AccountName=demoaccount;AccountKey=QdcckhAuPxEPKOSrw6vGv4MG+Nqm+tOuLOMncQfS0tYLl05auuo00zemdO70EMgTkAZhAHR7eUwHZA31K2oLQQ==;EndpointSuffix=core.windows.net"; "rdb-backup-frequency" = "30"}

Reviewer: Is the account key yours?

> This command creates a Redis Cache.

Reviewer 2: ```suggestion
This cmdlet creates a cache using Azure Cache for Redis.
```

```
PS C:\>New-AzRedisCache -ResourceGroupName "MyGroup" -Name "MyCache" -Location "Central US" -Size P1 -Sku "Premium" -Zone @("1","2")

          PrimaryKey         : pJ+jruGKPHDKsEC8kmoybobH3TZx2njBR3ipEsquZFo=
          SecondaryKey       : sJ+jruGKPHDKsEC8kmoybobH3TZx2njBR3ipEsquZFo=
          ResourceGroupName  : MyGroup
          Id                 : /subscriptions/a559b6fd-3a84-40bb-a450-b0db5ed37dfe/resourceGroups/mygroup/providers/Microsoft.Cache/Redis/MyCache
          Location           : Central US
          Name               : mycache
          Type               : Microsoft.Cache/Redis
          HostName           : mycache.redis.cache.windows.net
          Port               : 6379
          ProvisioningState  : creating
          SslPort            : 6380
          RedisConfiguration : {[maxmemory-policy, allkeys-random], [maxclients, 7500], [maxmemory-reserved, 200],
                                [maxfragmentationmemory-reserved, 300]...} 
          EnableNonSslPort   : False
          RedisVersion       : 4.0.14
          Size               : 6GB
          Sku                : Premium
          Tag                : {}
          Zone               : {1, 2}
```

> This command creates Redis cache instance in mutliple zones.

Reviewer 2: ```suggestion
This cmdlet creates a cache instance in multiple zones.
```

          SslPort            : 6380
          RedisConfiguration : {[maxmemory-policy, allkeys-random], [maxclients, 7500], [maxmemory-reserved, 200],
                                [maxfragmentationmemory-reserved, 300]...} 
          EnableNonSslPort   : False
          RedisVersion       : 4.0.14
          Size               : 6GB
          Sku                : Premium
          SubnetId           : /subscriptions/a559b6fd-3a84-40bb-a450-b0db5ed37dfe/resourceGroups/mygroup/providers/Microsoft.Network/virtualNetworks/MyNet/subnets/MySubnet
          StaticIP           : 10.0.0.4  
          Tag                : {}
          Zone               : []
```

### Example 5: Configure data persistence for a Premium Azure Cache for Redis

Persistence writes Redis data into an Azure Storage account that you own and manage. So before configuring data persistence you need to have [storage account](https://docs.microsoft.com/en-us/azure/storage/common/storage-account-create?tabs=azure-powershell) in same resource group. Choose a storage account in the same region and subscription as the cache, and a Premium Storage account is recommended because premium storage has higher throughput.

> After creating Storage account get storage account connection string. Steps to getting connection string from storage account.

Reviewer 2: ```suggestion
After creating a storage account, get the storage account connection string using this procedure. 
```

1. Run this command ** "Get-AzStorageAccountKey -ResourceGroupName $resourceGroupName -Name $storageAccountName"** in powershell.

> 1. From the output of above command copy any key.

Reviewer 2: ```suggestion
1. From the output of above, copy any key.
```

> 1. Put the storage account key and the storage account name in below format to get connection string of Storage account.

Reviewer 2: ```suggestion
1. Put the storage account key and the storage account name in format below to get the connection string of your storage account.
```

```
Connection String Format :- "DefaultEndpointsProtocol=https;AccountName={storageAccountName};AccountKey={storageAccountKey};EndpointSuffix=core.windows.net"
```

> There is need to specifies Redis configuration settings to enable data persistence. 

Reviewer 2: ```suggestion
You must have the specific Redis configuration settings to enable data persistence. 
```
